helix-term/src/commands.rs

```diff
@@ -2266,6 +2286,42 @@ fn search_selection(cx: &mut Context) {
     }
 }
 
+fn is_bow(index: usize, contents: RopeSlice) -> bool {
+    match index.checked_sub(1) {
+        Some(prev_index) => get_successive_chars(prev_index, contents)
+            .map(|(c1, c2)| !char_is_word(c1) && char_is_word(c2))
+            .unwrap_or(false),
+        // we are at the beginning of the file
+        None => contents.get_char(index).map(char_is_word).unwrap_or(false),
+    }
+}
+
+fn is_eow(index: usize, contents: RopeSlice) -> bool {
+    match index.checked_sub(1) {
+        Some(prev_index) => {
+            // we are at the end of the file
+            if index == contents.len_chars() {
+                contents
+                    .get_char(prev_index)
+                    .map(char_is_word)
+                    .unwrap_or(false)
+            } else {
+                get_successive_chars(prev_index, contents)
+                    .map(|(c1, c2)| char_is_word(c1) && !char_is_word(c2))
+                    .unwrap_or(false)
+            }
+        }
+        None => false,
+    }
+}
```

Comment:
These can be moved into the body of `search_selection_impl`. I don't think we will need them elsewhere so I don't think we need to move them somewhere generic like `helix-stdx/src/rope.rs`

Reply:
Done